services/api/src/models/Quests.ts

```diff
@@ -41,7 +42,9 @@ export function searchQuests(db: Database, userId: string, params: QuestSearchPa
   // TODO: Validate search params
   const where: Sequelize.WhereOptions<Partial<Quest>> = {published: {$ne: null} as any, tombstone: null};
 
-  where.partition = params.partition || PUBLIC_PARTITION;
+  if (params.showPrivate === false) {
+    where.partition = params.partition || PUBLIC_PARTITION;
```

Comment:
If showPrivate is true, we should explicitly only search for quests where partition is PUBLIC_PARTITION or PRIVATE_PARTITION, since we may add future partitions in the future. Link for the Sequelize `Op.or` syntax: http://docs.sequelizejs.com/manual/tutorial/querying.html

We should also make sure to sort by partition first, so that private quests are listed first, if showPrivate is true.

Reply:
On it.